components/drop-down.js

```diff
@@ -0,0 +1,176 @@
+import { Components } from "./base.js";
+
+const hiddenClass = "hidden";
+
+const defaultContainerClass = "cursor-pointer text-xl w-fit";
+const defaultTitleClass = "text-xl py-2";
+const defaultOptionsClass = "w-full space-y-1 overflow-auto max-h-[90vh]";
+const defaultOptionClass = "py-2 px-2 text-right rounded bg-slate-50";
+
+const attributes = {
+    hideOptionsOnOutsideClick: "hide-options-on-outside-click",
+    hideOptionsOnSelected: "hide-options-on-selected",
+    optionsZIndex: "options-z-index"
+};
+
+const optionDataId = "data-id";
+
+const elements = {
+    container: "container",
+    title: "title",
+    options: "options",
+    option: "option",
+    optionOfIdx(idx) {
+        return `option-${idx}`
+    }
+};
+
+const maxOptions = 100;
+
+class DropDown extends HTMLElement {
+
+    set options(options) {
```

Comment:
I've added this setter to make it easy to define options on this element both in JS and in HTML:
```
// HTML
 <drop-down title="HTMX & attributes-based Drop-down"
            hide-options-on-outside-click="false"
            hide-options-on-selected="false"
            options-z-index="100"
            id="attributes-drop-down"
            option:add:class="text-2xl"
            option-0-text="A"
            option-0:add:class="font-bold"
            option-0:data-id="A-1"
            option-0:hx-get="${PATH}/option-0"
            option-0:hx-push-url="true"
            option-0:hx-target="#app"
            option-1-text="B"
            option-1:data-id="B-1">
        </drop-down>
        
// JS
const jsDropdown = document.getElementById("js-drop-down");  
jsDropdown.setAttribute("option-0:add:class", "text-2xl");
const options = [
    {
        title: "Option A",
        dataId: 22
    },
    {
        title: "Option B"
     }
 ];
    
jsDropdown.options = options;
```